plugins/kubernetes/app/models/kubernetes/resource.rb

```diff
@@ -38,9 +38,17 @@ def revert(previous)
         end
       end
 
+      # wait for delete to finish before doing further work so we don't run into duplication errors
+      # sum of waits should be ~30s which is the default delete timeout
       def delete
-        request(:delete, name, namespace)
-        expire_cache
+        return true unless running?
+        request_delete
+        [0.0, 0.1, 0.2, 0.5, 1, 2, 4, 8, 16].each do |wait|
```

Comment:
first wait is 0 since the request itself already takes a few ms and might be long enough